connect_transformations/webapp.py

```diff
@@ -164,6 +166,81 @@ def validate_currency_conversion(self, data):
             'overview': overview,
         }
 
+    def validate_split_column(self, data):
+        if (
+            'settings' not in data
+            or not type(data['settings']) == dict
+            or 'columns' not in data
+            or 'input' not in data['columns']
+        ):
+            return JSONResponse(status_code=400, content={'error': 'Invalid input data'})
+
+        if (
+            'from' not in data['settings']
+            or 'regex' not in data['settings']
+            or 'pattern' not in data['settings']['regex']
+            or not type(data['settings']['regex']['pattern']) == str
+            or 'groups' not in data['settings']['regex']
+            or not type(data['settings']['regex']['groups']) == dict
+            or 'order' not in data['settings']['regex']
+            or not type(data['settings']['regex']['order']) == list
+        ):
+            return JSONResponse(
+                status_code=400,
+                content={
+                    'error': (
+                        'The settings must have `from` and `regex` with `pattern`, `groups` and '
+                        '`order` fields'
+                    ),
+                },
+            )
+
+        input_columns = data['columns']['input']
+        available_input_columns = [c['name'] for c in input_columns]
+        if data['settings']['from'] not in available_input_columns:
+            return JSONResponse(
+                status_code=400,
+                content={
+                    'error': (
+                        'The settings contains an invalid `from` column name'
+                        f' "{data["settings"]["from"]}" that does not exist on '
+                        'columns.input'
+                    ),
+                },
+            )
+
+        groups = data['settings']['regex']['groups']
+        for key in groups.keys():
```

Comment:
you can get the list of named capturing group compiling the regex and using https://docs.python.org/3/library/re.html#re.Pattern.groupindex